src/hearings/containers/request-hearing/hearing-link/hearing-link.component.ts

```diff
@@ -40,14 +44,30 @@ export class HearingLinkComponent extends RequestHearingPageFlow implements OnIn
       caseReference: this.caseId,
       hearingId: ''
     }));
-    this.hearingLinksSub = this.hearingStore.pipe(select(fromHearingStore.getHearingLinks)).subscribe(
-      hearingLinks => {
-        if (hearingLinks.serviceLinkedCases) {
-          this.linkedCases = hearingLinks.serviceLinkedCases;
-          this.showSpinner = false;
+    this.generateLinkedCasesWithReasonDescription();
+  }
+
+  public generateLinkedCasesWithReasonDescription(): void {
+    this.hearingsServiceSub = this.hearingsService.loadCaseLinkingReasonCodes().subscribe(reasons => {
+      this.hearingLinksSub = this.hearingStore.pipe(select(fromHearingStore.getHearingLinks)).subscribe(
+        hearingLinks => {
+          if (hearingLinks.serviceLinkedCases) {
+            hearingLinks.serviceLinkedCases.forEach(linkedCase => {
+              const caseLinkingReasons = reasons.list_of_values.filter(reason => linkedCase.reasonsForLink.some(reasonCode => reason.key === reasonCode));
+              const caseLinkingReasonsValues = caseLinkingReasons.map(x => x.value_en);
+              if (caseLinkingReasonsValues && caseLinkingReasonsValues.length > 0) {
+                this.linkedCases.push({caseName: linkedCase.caseName, caseReference: linkedCase.caseReference, reasonsForLink: caseLinkingReasonsValues});
+              } else {
+                this.linkedCases.push(linkedCase);
+              }
+            });
+            this.showSpinner = false;
+          }
         }
-      }
-    );
+      );
+    }, () => {
+      this.router.navigate(['/hearings/error']);
```

Comment:
remove unnecessary bracing

`() => this.router.navigate(['/hearings/error']);`

Reply:
This will be done as part of the PR https://github.com/hmcts/rpx-xui-webapp/pull/2630 which you already commented on and merged later into this.